resources/smpp/ra/src/main/java/org/restcomm/slee/resource/smpp/SmppSessionsImpl.java

```diff
@@ -180,16 +160,33 @@ private void countSendResponsePdu(SmppSessionCounters counters, PduResponse pdu,
             // TODO: adding here statistics for SUBMIT_MULTI ?
             }
         }
-    }	
+    }
 
-	protected class SmppSessionHandlerInterfaceImpl implements SmppSessionHandlerInterface {
+	protected class SmppSessionHandlerInterfaceImpl implements
+			SmppSessionHandlerInterface {
 
 		public SmppSessionHandlerInterfaceImpl() {
 
 		}
 
+		public void destroySmppSessionHandler(Esme esme) {
+			if (esme != null && esme.getName() != null) {
+				SenderThread senderThread = esmeSenderThreads.remove(esme
+						.getName());
+				if (senderThread != null)
+					senderThread.deactivate();
+			}
+		}
+
 		@Override
 		public SmppSessionHandler createNewSmppSessionHandler(Esme esme) {
+
+			SenderThread senderThread = new SenderThread("SMPP ESME Sender " + esme.getName());
```

Comment:
is the creation of SenderThread light enough to support putIfAbsent technique? that is, be optimal on cpu/mem consumption if several threads try to insert the thread into the map...